From 557ce4b7dd9204696fb0992a47123721470a407a Mon Sep 17 00:00:00 2001
From: Krzesimir Nowak <knowak@microsoft.com>
Date: Thu, 20 Nov 2025 13:16:09 +0100
Subject: [PATCH 1/2] [build] Simplify installation of symlinks and manpages

Signed-off-by: Krzesimir Nowak <knowak@microsoft.com>
---
 Makefile | 52 ++++++++--------------------------------------------
 1 file changed, 8 insertions(+), 44 deletions(-)

diff --git a/Makefile b/Makefile
index b04937aa..44c99c99 100644
--- a/Makefile
+++ b/Makefile
@@ -55,55 +55,19 @@ TOOLS:=\
 	era_invalidate \
 	era_restore
 
+# This must be two empty lines to get a newline.
+define NEWLINE
+
+
+endef
+
 MANPAGES:=$(patsubst %,man8/%.8,$(TOOLS))
 
 install: $(MANPAGES)
 	$(INSTALL_DIR) $(BINDIR)
 	$(INSTALL_PROGRAM) $(PDATA_TOOLS) $(BINDIR)
-	ln -s -f pdata_tools $(BINDIR)/cache_check
-	ln -s -f pdata_tools $(BINDIR)/cache_dump
-	ln -s -f pdata_tools $(BINDIR)/cache_metadata_size
-	ln -s -f pdata_tools $(BINDIR)/cache_repair
-	ln -s -f pdata_tools $(BINDIR)/cache_restore
-	ln -s -f pdata_tools $(BINDIR)/cache_writeback
-	ln -s -f pdata_tools $(BINDIR)/thin_check
-	ln -s -f pdata_tools $(BINDIR)/thin_delta
-	ln -s -f pdata_tools $(BINDIR)/thin_dump
-	ln -s -f pdata_tools $(BINDIR)/thin_ls
-	ln -s -f pdata_tools $(BINDIR)/thin_repair
-	ln -s -f pdata_tools $(BINDIR)/thin_restore
-	ln -s -f pdata_tools $(BINDIR)/thin_rmap
-	ln -s -f pdata_tools $(BINDIR)/thin_metadata_size
-	ln -s -f pdata_tools $(BINDIR)/thin_metadata_pack
-	ln -s -f pdata_tools $(BINDIR)/thin_metadata_unpack
-	ln -s -f pdata_tools $(BINDIR)/thin_migrate
-	ln -s -f pdata_tools $(BINDIR)/thin_trim
-	ln -s -f pdata_tools $(BINDIR)/era_check
-	ln -s -f pdata_tools $(BINDIR)/era_dump
-	ln -s -f pdata_tools $(BINDIR)/era_invalidate
-	ln -s -f pdata_tools $(BINDIR)/era_restore
+	$(foreach tool, $(TOOLS), ln -s -f pdata_tools $(BINDIR)/$(tool); $(NEWLINE))
 	$(INSTALL_DIR) $(MANPATH)/man8
-	$(INSTALL_DATA) man8/cache_check.8 $(MANPATH)/man8
-	$(INSTALL_DATA) man8/cache_dump.8 $(MANPATH)/man8
-	$(INSTALL_DATA) man8/cache_metadata_size.8 $(MANPATH)/man8
-	$(INSTALL_DATA) man8/cache_repair.8 $(MANPATH)/man8
-	$(INSTALL_DATA) man8/cache_restore.8 $(MANPATH)/man8
-	$(INSTALL_DATA) man8/cache_writeback.8 $(MANPATH)/man8
-	$(INSTALL_DATA) man8/thin_check.8 $(MANPATH)/man8
-	$(INSTALL_DATA) man8/thin_delta.8 $(MANPATH)/man8
-	$(INSTALL_DATA) man8/thin_dump.8 $(MANPATH)/man8
-	$(INSTALL_DATA) man8/thin_ls.8 $(MANPATH)/man8
-	$(INSTALL_DATA) man8/thin_repair.8 $(MANPATH)/man8
-	$(INSTALL_DATA) man8/thin_restore.8 $(MANPATH)/man8
-	$(INSTALL_DATA) man8/thin_rmap.8 $(MANPATH)/man8
-	$(INSTALL_DATA) man8/thin_metadata_size.8 $(MANPATH)/man8
-	$(INSTALL_DATA) man8/thin_metadata_pack.8 $(MANPATH)/man8
-	$(INSTALL_DATA) man8/thin_metadata_unpack.8 $(MANPATH)/man8
-	$(INSTALL_DATA) man8/thin_migrate.8 $(MANPATH)/man8
-	$(INSTALL_DATA) man8/era_check.8 $(MANPATH)/man8
-	$(INSTALL_DATA) man8/era_dump.8 $(MANPATH)/man8
-	$(INSTALL_DATA) man8/era_restore.8 $(MANPATH)/man8
-	$(INSTALL_DATA) man8/era_invalidate.8 $(MANPATH)/man8
-	$(INSTALL_DATA) man8/thin_trim.8 $(MANPATH)/man8
+	$(foreach tool, $(TOOLS), $(INSTALL_DATA) man8/$(tool).8 $(MANPATH)/man8; $(NEWLINE))
 
 .PHONY: install
-- 
2.52.0

