From b45ba164aadfbfd5b1137d57ca5c800d9ec4ed6a Mon Sep 17 00:00:00 2001
Message-Id: <b45ba164aadfbfd5b1137d57ca5c800d9ec4ed6a.1574953996.git.dongsu@kinvolk.io>
From: Dongsu Park <dongsu@kinvolk.io>
Date: Thu, 28 Nov 2019 16:12:41 +0100
Subject: [PATCH] components/engine: fix name of gcc binary for aarch64

---
 components/engine/hack/make/.binary | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/hack/make/.binary b/hack/make/.binary
index 53de6749..3c5307ba 100644
--- a/hack/make/.binary
+++ b/hack/make/.binary
@@ -60,7 +60,7 @@ if [ "$(go env GOOS)/$(go env GOARCH)" != "$(go env GOHOSTOS)/$(go env GOHOSTARC
 			esac
 			;;
 		linux/arm64)
-			export CC=aarch64-linux-gnu-gcc
+			export CC=aarch64-cros-linux-gnu-gcc
 			export CGO_ENABLED=1
 			;;
 		linux/amd64)
-- 
2.23.0

