fix build failure on hppa

--- src/xdr_float.c
+++ src/xdr_float.c
@@ -59,7 +59,7 @@
     defined(__arm32__) || defined(__ppc__) || defined(__ia64__) || \
     defined(__arm26__) || defined(__sparc64__) || defined(__amd64__) || \
     defined(__powerpc__) || defined(__s390__) || defined(__arm__) || \
-    defined(__sh__)
+    defined(__sh__) || defined(__hppa__)
 #include <bits/endian.h>
 #define IEEEFP
 #endif
