diff -ur texinfo-6.8/Makefile.am texinfo-6.8-patched/Makefile.am
--- texinfo-6.8/Makefile.am	2021-02-16 21:07:41.000000000 +0000
+++ texinfo-6.8-patched/Makefile.am	2021-09-21 11:05:15.495497278 +0000
@@ -62,9 +62,7 @@
   SUBDIRS += info
 endif
 endif
-  SUBDIRS += install-info po po_document tp Pod-Simple-Texinfo texindex util \
-             doc man
-  SUBDIRS += js
+  SUBDIRS += install-info po po_document tp Pod-Simple-Texinfo texindex util
 endif
 
 distclean-local:
diff -ur texinfo-6.8/configure.ac texinfo-6.8-patched/configure.ac
--- texinfo-6.8/configure.ac	2021-07-03 09:12:42.000000000 +0000
+++ texinfo-6.8-patched/configure.ac	2021-09-21 11:03:11.238623956 +0000
@@ -192,7 +192,7 @@
   # env -i gives this build host configure a clean environment;
   # consequently, we have to re-initialize $PATH.
   env -i CC="$BUILD_CC" AR="$BUILD_AR" RANLIB="$BUILD_RANLIB" \
-      PATH="$PATH" \
+      PATH="$PATH" CCACHE_DIR="$CCACHE_DIR" \
       tools_only=1 \
     ${confdir}/configure --build=${build} --host=${build} \
                          --disable-rpath --disable-nls
