pam_module: use /var/lib/ instead of /var

--- a/src/oslogin_utils.cc
+++ b/src/oslogin_utils.cc
@@ -57,8 +57,8 @@
 
 // Regex for validating user names.
 static const char kUserNameRegex[] = "^[a-zA-Z0-9._][a-zA-Z0-9._-]{0,31}$";
-static const char kSudoersDir[] = "/var/google-sudoers.d/";
-static const char kUsersDir[] = "/var/google-users.d/";
+static const char kSudoersDir[] = "/var/lib/google-sudoers.d/";
+static const char kUsersDir[] = "/var/lib/google-users.d/";
 
 namespace oslogin_utils {
 
