http://bugs.gentoo.org/193920

--- psmisc-22.5/Makefile.am
+++ psmisc-22.5/Makefile.am
@@ -1,5 +1,5 @@
 
-SUBDIRS = doc src po icons 
+SUBDIRS = doc src icons 
 
 #EXTRA_DIST = config.rpath ABOUT-NLS mkinstalldirs
 
--- psmisc-22.5/configure.ac
+++ psmisc-22.5/configure.ac
@@ -65,10 +65,6 @@
    test $ac_cv_member_struct_user_regs_struct_ecx = yes &&
    test $ac_cv_member_struct_user_regs_struct_edx = yes )
 
-dnl Check for language stuff
-AM_GNU_GETTEXT([external])
-AM_GNU_GETTEXT_VERSION([0.16.1])
-
 dnl Checks for library functions.
 AC_FUNC_CLOSEDIR_VOID
 AC_FUNC_GETMNTENT
@@ -87,6 +83,5 @@
 AC_CONFIG_FILES([Makefile
 		 doc/Makefile
 		 src/Makefile
-		 icons/Makefile
-		 po/Makefile.in])
+		 icons/Makefile])
 AC_OUTPUT
