diff --git a/agent/plugins/inventory/gatherers/application/dataProvider.go b/agent/plugins/inventory/gatherers/application/dataProvider.go
index d252ffb4..d1a010ed 100644
--- a/agent/plugins/inventory/gatherers/application/dataProvider.go
+++ b/agent/plugins/inventory/gatherers/application/dataProvider.go
@@ -7,7 +7,6 @@ import (
 	"fmt"

 	"github.com/aws/amazon-ssm-agent/agent/context"
-	"github.com/aws/amazon-ssm-agent/agent/plugins/configurepackage/localpackages"
 	"github.com/aws/amazon-ssm-agent/agent/plugins/inventory/model"
 )

diff --git a/agent/plugins/inventory/gatherers/application/dataProvider_unix_test.go b/agent/plugins/inventory/gatherers/application/dataProvider_unix_test.go
index af51fa2d..a2757005 100644
--- a/agent/plugins/inventory/gatherers/application/dataProvider_unix_test.go
+++ b/agent/plugins/inventory/gatherers/application/dataProvider_unix_test.go
@@ -241,4 +241,4 @@ func TestApplicationDataWithPackageRepositoryData(t *testing.T) {
 	data := CollectApplicationData(mockContext)
 	assert.Equal(t, len(sampleDataParsed), len(data))
 	assert.NotEqual(t, len(mockData), len(data))
-}
\ No newline at end of file
+}
