--- a/Configure
+++ b/Configure
@@ -2294,3 +2294,3 @@
     fi	# }
-    LSOF_CFGF="-DLINUXV=$LSOF_VERS"
+    LSOF_CFGF="${CFLAGS} -DLINUXV=$LSOF_VERS"
     if test $LSOF_VERS -lt 21072	# {
@@ -4828,3 +4828,3 @@
   echo "" >> $LSOF_MKFC
-  echo "CFGL=	$LSOF_FCFGL -L./$LSOF_LIB -llsof $LSOF_CFGL" >> $LSOF_MKFC
+  echo "CFGL=	$LSOF_FCFGL -L./$LSOF_LIB -llsof $LSOF_CFGL ${LDFLAGS}" >> $LSOF_MKFC
 fi	# }
--- a/Customize
+++ b/Customize
@@ -110,7 +110,7 @@
 do
   echo ""
   echo $EO "Do you want to customize (y|n) [y]? $EC"
-  read ANS EXCESS
+  ANS=y
   if test "X$ANS" = "Xn" -o "X$ANS" = "XN"
   then
     exit 0
@@ -214,7 +214,7 @@
   else
     echo $EO "Enable HASSECURITY (y|n) [n]? $EC"
   fi
-  read ANS EXCESS
+  ANS=y
   if test "X$ANS" = "Xy" -o "X$ANS" = "XY"
   then
     echo ""
@@ -277,7 +277,7 @@
     else
       echo $EO "Enable HASNOSOCKSECURITY (y|n) [n]? $EC"
     fi
-    read ANS EXCESS
+    ANS=y
     if test "X$ANS" = "Xy" -o "X$ANS" = "XY"
     then
       echo ""
@@ -339,7 +339,7 @@
   else
     echo $EO "Disable WARNINGSTATE? (y|n) [n]? $EC"
   fi
-  read ANS EXCESS
+  ANS=n
   if test "X$ANS" = "Xy" -o "X$ANS" = "XY"
   then
     echo ""
@@ -408,7 +408,7 @@
     else
       echo $EO "Enable WARNDEVACCESS (y|n) [n]? $EC"
     fi
-    read ANS EXCESS
+    ANS=n
     if test "X$ANS" = "Xy" -o "X$ANS" = "XY"
     then
       echo ""
@@ -491,7 +491,7 @@
     else
       echo $EO "Enable HASDCACHE (y|n) [n]? $EC"
     fi
-    read ANS EXCESS
+    ANS=n
     if test "X$ANS" = "Xy" -o "X$ANS" = "XY"
     then
       echo ""
@@ -545,7 +545,7 @@
     do
       echo ""
       echo $EO "Do you want to change path options (y|n) [n]? $EC"
-      read ANS EXCESS
+      ANS=n
       if test "X$ANS" = "Xy" -o "X$ANS" = "XY"
       then
         CDC=1
@@ -928,7 +928,7 @@
       else
         echo $EO "Enable HASKERNIDCK (y|n) [n]? $EC"
       fi
-      read ANS EXCESS
+      ANS=y
       if test "X$ANS" = "Xy" -o "X$ANS" = "XY"
       then
         echo ""
@@ -1125,7 +1125,7 @@
   echo ""
   echo "Do you want to rename $OLD to ${OLD}.old and replace it with"
   echo $EO "$NEW (y|n) [y]? $EC"
-  read ANS EXCESS
+  ANS=y
   if test "X$ANS" = "Xn" -o "X$ANS" = "XN"
   then
     END=1
