http://krbdev.mit.edu/rt/Ticket/Display.html?id=7000

Index: trunk/src/kadmin/server/ipropd_svc.c
===================================================================
diff -u -N -r24961 -r25433
--- trunk/src/kadmin/server/ipropd_svc.c	(.../ipropd_svc.c)	(revision 24961)
+++ trunk/src/kadmin/server/ipropd_svc.c	(.../ipropd_svc.c)	(revision 25433)
@@ -380,7 +380,7 @@
 			     _("%s: pclose(popen) failed: %s"),
 			     whoami,
 			     error_message(errno));
-	    goto out;
+	    _exit(1);
 	}
 
 	DPRINT(("%s: exec `kprop -f %s %s' ...\n",
@@ -401,7 +401,7 @@
 			     _("%s: exec failed: %s"),
 			     whoami,
 			     error_message(errno));
-	    goto out;
+	    _exit(1);
 	}
 
     default: /* parent */
