Index: java_config_2/EnvironmentManager.py
===================================================================
--- src/java_config_2/EnvironmentManager.py	(revision 7809)
+++ src/java_config_2/EnvironmentManager.py	(working copy)
@@ -81,7 +81,7 @@
                 raise UnexistingPackageError(name)
 
     def load_packages(self):
-        for package in iter(glob(self.pkg_path) % "*" ):
+        for package in iter(glob(self.pkg_path % "*" )):
             name = basename(dirname(package))
             if name in self.packages:
                 continue
