--- numpy/distutils/ccompiler.py
+++ numpy/distutils/ccompiler.py
@@ -2,9 +2,11 @@
 import os
 import sys
 import new
+from copy import copy
 
 from distutils.ccompiler import *
 from distutils import ccompiler
+from distutils.errors import DistutilsExecError, DistutilsModuleError, DistutilsPlatformError
 from distutils.sysconfig import customize_compiler
 from distutils.version import LooseVersion
