diff -Naur PyYAML-3.05.orig/setup.py PyYAML-3.05/setup.py
--- PyYAML-3.05.orig/setup.py	2007-05-12 23:28:55.000000000 +0200
+++ PyYAML-3.05/setup.py	2007-10-16 02:29:57.000000000 +0200
@@ -31,6 +31,8 @@
 ]
 
 from distutils.core import setup
+from distutils.extension import Extension
+from Pyrex.Distutils import build_ext
 
 if __name__ == '__main__':
 
@@ -49,5 +51,10 @@
 
         package_dir={'': 'lib'},
         packages=['yaml'],
+        ext_modules=[
+            Extension("_yaml", ["ext/_yaml.pyx"], libraries=['yaml']),
+        ],
+
+        cmdclass = {'build_ext': build_ext}
     )
 
