--- src/freeglut_cursor.c.old	2006-10-11 20:49:13.000000000 +0200
+++ src/freeglut_cursor.c	2006-10-11 20:51:43.000000000 +0200
@@ -147,11 +147,13 @@
             }
         }
 
-        if ( ( cursorIDToUse != GLUT_CURSOR_NONE ) && ( cursor == None ) ) {
+	if ( cursorIDToUse == GLUT_CURSOR_INHERIT ) {
+            XUndefineCursor( fgDisplay.Display, window->Window.Handle );
+	} else if ( cursor != None ) {
+            XDefineCursor( fgDisplay.Display, window->Window.Handle, cursor );
+	} else if ( cursorIDToUse != GLUT_CURSOR_NONE ) {
             fgError( "Failed to create cursor" );
         }
-        XDefineCursor( fgDisplay.Display,
-                       window->Window.Handle, cursor );
     }
 
 #elif TARGET_HOST_WIN32 || TARGET_HOST_WINCE
