diff -aur iniparser3.0b/src/iniparser.h iniparser3.0b.patched/src/iniparser.h
--- iniparser3.0b/src/iniparser.h	2007-11-23 21:38:19.000000000 +0000
+++ iniparser3.0b.patched/src/iniparser.h	2009-04-28 11:08:28.644706261 +0100
@@ -39,7 +39,8 @@
  ---------------------------------------------------------------------------*/
 /** For backwards compatibility only */
 #define iniparser_getstr(d, k)  iniparser_getstring(d, k, NULL)
-#define iniparser_setstr        iniparser_setstring
+#define iniparser_setstr        iniparser_set
+#define iniparser_setstring     iniparser_set
 
 /*-------------------------------------------------------------------------*/
 /**
@@ -215,11 +216,11 @@
   @return   int 0 if Ok, -1 otherwise.
 
   If the given entry can be found in the dictionary, it is modified to
-  contain the provided value. If it cannot be found, -1 is returned.
+  contain the provided value. If it cannot be found, it is inserted.
   It is Ok to set val to NULL.
  */
 /*--------------------------------------------------------------------------*/
-int iniparser_setstring(dictionary * ini, char * entry, char * val);
+int iniparser_set(dictionary * ini, char * entry, char * val);
 
 
 /*-------------------------------------------------------------------------*/
