diff -Naur a/gcc/gcc.cc b/gcc/gcc.cc
--- a/gcc/gcc.cc	2024-08-01 23:34:33.525082176 +0100
+++ b/gcc/gcc.cc	2024-08-01 23:43:31.557156041 +0100
@@ -5527,6 +5527,16 @@
 	      "BINUTILS", PREFIX_PRIORITY_LAST, 0, 1);
   free (tooldir_prefix);

+  if (*cross_compile == '1' && !target_system_root_changed)
+    {
+      const char *esysroot = env.get("ESYSROOT");
+      if (esysroot && esysroot[0] != '\0' && strcmp(esysroot, "/") != 0 && (!target_system_root || strcmp(esysroot, target_system_root) != 0))
+	{
+	  target_system_root = esysroot;
+	  target_system_root_changed = 1;
+	}
+    }
+
 #if defined(TARGET_SYSTEM_ROOT_RELOCATABLE) && !defined(VMS)
   /* If the normal TARGET_SYSTEM_ROOT is inside of $exec_prefix,
      then consider it to relocate with the rest of the GCC installation
