From 1fef848147fb1e09a96dbebee5fdc1760519c259 Mon Sep 17 00:00:00 2001
From: Krzesimir Nowak <knowak@microsoft.com>
Date: Thu, 6 Jan 2022 14:06:04 +0100
Subject: [PATCH] configure: specify gettext version

---
 configure.ac | 1 +
 1 file changed, 1 insertion(+)

diff --git a/configure.ac b/configure.ac
index edd184154..d661f1446 100644
--- a/configure.ac
+++ b/configure.ac
@@ -343,6 +343,7 @@ AC_CHECK_PROG(HAVE_CXX, $CXX, yes, no)
 
 AC_GNU_SOURCE
 AM_GNU_GETTEXT([external])
+AM_GNU_GETTEXT_VERSION([0.20])
 AC_SYS_LARGEFILE
 
 # Identify characteristics of the host architecture.
-- 
2.34.1

