From 7fda98ef4a405fdfe6f2f951398197a573ca7077 Mon Sep 17 00:00:00 2001
From: Krzesimir Nowak <knowak@microsoft.com>
Date: Fri, 10 Oct 2025 11:47:43 +0200
Subject: [PATCH 2/2] Flatcar modifications

---
 templates/system-auth.tpl | 24 +++++++++++++++---------
 1 file changed, 15 insertions(+), 9 deletions(-)

diff --git a/templates/system-auth.tpl b/templates/system-auth.tpl
index 905d04f..b211abb 100644
--- a/templates/system-auth.tpl
+++ b/templates/system-auth.tpl
@@ -9,11 +9,15 @@ auth		[default={{ 3 + homed + (sssd * 3) }}]	pam_permit.so
 {% endif %}
 
 {% if sssd %}
-auth		[default=1 ignore=ignore success=ok]	pam_usertype.so isregular
-auth		[default=3 ignore=ignore success=ok]	pam_localuser.so
+# FLATCAR: Removed. We aren't limiting login to regular users.
+# auth		[default=1 ignore=ignore success=ok]	pam_usertype.so isregular
+# FLATCAR: Removed. We have some users and groups in
+# /usr/share/baselayout/passwd, not only in /etc/passwd.
+# auth		[default=3 ignore=ignore success=ok]	pam_localuser.so
 {% endif %}
 
-auth		requisite	pam_faillock.so preauth
+# FLATCAR: Added deny, unlock_time and fail_interval to override defaults.
+auth		requisite	pam_faillock.so preauth preauth deny=5 unlock_time=60 fail_interval=120
 
 {% if homed %}
 auth		[success=2 default=ignore]	pam_systemd_home.so
@@ -21,13 +25,11 @@ auth		[success=2 default=ignore]	pam_systemd_home.so
 
 {% if sssd %}
 auth		sufficient	pam_unix.so {{ nullok }} {{ debug }}
+auth		sufficient	pam_sss.so forward_pass {{ debug }}
 {% else %}
 auth		[success=1 new_authtok_reqd=1 ignore=ignore default=bad]	pam_unix.so {{ nullok }} {{ debug }} try_first_pass
 {% endif %}
 auth		[default=die]	pam_faillock.so authfail
-{% if sssd %}
-auth		sufficient	pam_sss.so forward_pass {{ debug }}
-{% endif %}
 {% if caps %}
 auth		optional	pam_cap.so
 {% endif %}
@@ -45,9 +47,13 @@ account		[success={{ 2 if sssd else 1 }} default=ignore]	pam_systemd_home.so
 account		required	pam_unix.so {{ debug }}
 account		required	pam_faillock.so
 {% if sssd %}
-account		sufficient	pam_localuser.so
-account		sufficient	pam_usertype.so issystem
-account		[default=bad success=ok user_unknown=ignore]	pam_sss.so {{ debug }}
+# FLATCAR: Removed. We have some users and groups in
+# /usr/share/baselayout/passwd, not only in /etc/passwd.
+# account		sufficient	pam_localuser.so
+# FLATCAR: Removed. Maybe we could keep it?
+# account		sufficient	pam_usertype.so issystem
+# FLATCAR: Added ignore when sssd is not running
+account		[default=bad success=ok user_unknown=ignore authinfo_unavail=ignore]	pam_sss.so {{ debug }}
 account		required	pam_permit.so
 {% endif %}
 
-- 
2.52.0

