--- setup.py
+++ setup.py
@@ -4,6 +4,7 @@
 import distutils.dir_util  as dut
 import distutils.file_util as fut
 import io
+import os
 import subprocess
 import sys
 
@@ -106,5 +107,5 @@
   description = "python interface for cairo",
   ext_modules = [cairo],
   data_files=[('include/pycairo',['src/pycairo.h']),
-              ('lib/pkgconfig',[pkgconfig_file])],
+              (os.environ.get('PKGCONFIG_DIR', 'lib/pkgconfig'),[pkgconfig_file])],
   )
