diff -r -u lrzsz-0.12.20/configure.in lrzsz-0.12.20-patched/configure.in
--- lrzsz-0.12.20/configure.in	2024-01-09 10:51:43.460543006 -0000
+++ lrzsz-0.12.20-patched/configure.in	2024-01-09 10:57:32.001193035 -0000
@@ -263,6 +263,11 @@
 AC_SUBST(LDFLAGS)
 AC_SUBST(LIBS)
 
+# a workaround for not AC_SUBSTed USE_INCLUDED_LIBINTL when disable-nls is passed
+if test "x${USE_NLS}" = xno ; then
+	USE_INCLUDED_LIBINTL=no
+	AC_SUBST(USE_INCLUDED_LIBINTL)
+fi
 
 AC_DEFINE_UNQUOTED(LOCALEDIR,"$prefix/$DATADIRNAME", [LOCALEDIR])
 AC_LINK_FILES($nls_cv_header_libgt, $nls_cv_header_intl)
