--- sqlite-3.6.22.orig/sqlite3.c	2010-01-13 22:28:39 -0800
+++ sqlite-3.6.22/sqlite3.c	2010-01-13 22:36:25 -0800
@@ -21915,7 +21915,9 @@
   case ENODEV:
   case ENXIO:
   case ENOENT:
+#ifdef ESTALE
   case ESTALE:
+#endif
   case ENOSYS:
     /* these should force the client to close the file and reconnect */
     
diff -ru sqlite-3.6.22.orig/sqlite3.c sqlite-3.6.22/sqlite3.c
--- sqlite-3.6.22.orig/sqlite3.c	2010-01-14 15:20:11 +0100
+++ sqlite-3.6.22/sqlite3.c	2010-01-14 15:21:25 +0100
@@ -23343,7 +23343,7 @@
     pFile->locktype = locktype;
 #if !OS_VXWORKS
     /* Always update the timestamp on the old file */
-    utimes(zLockFile, NULL);
+    utime(zLockFile, NULL);
 #endif
     return SQLITE_OK;
   }
