diff -ru sqlite-3.6.22/configure.ac sqlite-3.6.22.new/configure.ac
--- sqlite-3.6.22/configure.ac	2009-11-16 13:27:54.000000000 -0500
+++ sqlite-3.6.22.new/configure.ac	2010-01-13 12:21:27.092738446 -0500
@@ -637,6 +637,7 @@
       [use_loadextension=$enableval],[use_loadextension=no])
 if test "${use_loadextension}" = "yes" ; then
   OPT_FEATURE_FLAGS=""
+  AC_SEARCH_LIBS(dlopen, dl)
 else
   OPT_FEATURE_FLAGS="-DSQLITE_OMIT_LOAD_EXTENSION=1"
 fi
