diff -ru a/x11vnc/libvncserver/tightvnc-filetransfer/filetransfermsg.c b/x11vnc/libvncserver/tightvnc-filetransfer/filetransfermsg.c
--- a/x11vnc/libvncserver/tightvnc-filetransfer/filetransfermsg.c	2009-10-07 15:21:23.000000000 -0500
+++ b/x11vnc/libvncserver/tightvnc-filetransfer/filetransfermsg.c	2010-09-11 18:49:58.100074877 -0500
@@ -393,7 +393,7 @@
 CreateFileDownloadZeroSizeDataMsg(unsigned long mTime)
 {
 	FileTransferMsg fileDownloadZeroSizeDataMsg;
-	int length = sz_rfbFileDownloadDataMsg + sizeof(int);
+	int length = sz_rfbFileDownloadDataMsg + sizeof(unsigned long);
 	rfbFileDownloadDataMsg *pFDD = NULL;
 	char *pFollow = NULL;
 	
diff -ru a/x11vnc/x11vnc/remote.c b/x11vnc/x11vnc/remote.c
--- a/x11vnc/x11vnc/remote.c	2010-09-09 22:46:48.000000000 -0500
+++ b/x11vnc/x11vnc/remote.c	2010-09-11 18:18:37.948074878 -0500
@@ -5825,7 +5825,7 @@
 		}
 		if (!strcmp(p, "vncdisplay")) {
 			snprintf(buf, bufn, "aro=%s:%s", p,
-			    NONUL(vnc_desktop_name));
+			    vnc_desktop_name);
 			goto qry;
 		}
 		if (!strcmp(p, "icon_mode")) {
@@ -6027,7 +6027,7 @@
 		if (!strcmp(p, "h") || !strcmp(p, "help") ||
 		    !strcmp(p, "V") || !strcmp(p, "version") ||
 		    !strcmp(p, "lastmod")) {
-			snprintf(buf, bufn, "aro=%s:%s", p, NONUL(lastmod));
+			snprintf(buf, bufn, "aro=%s:%s", p, lastmod);
 			goto qry;
 		}
 		if (!strcmp(p, "bg")) {
diff -ru a/x11vnc/x11vnc/sslhelper.c b/x11vnc/x11vnc/sslhelper.c
--- a/x11vnc/x11vnc/sslhelper.c	2010-09-09 22:46:48.000000000 -0500
+++ b/x11vnc/x11vnc/sslhelper.c	2010-09-11 18:19:42.436074878 -0500
@@ -4004,7 +4004,7 @@
 				strncpy(last_get, rcookie, 100);
 				if (db) fprintf(stderr, "last_get: '%s'\n", last_get);
 			}
-			if (rcookie && strstr(rcookie, "VncViewer.class")) {
+			if (strstr(rcookie, "VncViewer.class")) {
 				rfbLog("\n");
 				rfbLog("helper[%d]:\n", pid);
 				rfbLog("***********************************************************\n");
diff -ru a/x11vnc/x11vnc/userinput.c b/x11vnc/x11vnc/userinput.c
--- a/x11vnc/x11vnc/userinput.c	2010-09-09 22:46:48.000000000 -0500
+++ b/x11vnc/x11vnc/userinput.c	2010-09-11 18:20:29.692074878 -0500
@@ -1305,7 +1305,7 @@
 	int font_size = 15;
 	int win_y, scr_y, loc_cut = 4*font_size, y_cut = 10*font_size;
 	
-	if (!xrecord_set_by_keys || !xrecord_name_info) {
+	if (!xrecord_set_by_keys) {
 		return 0;
 	}
 	if (xrecord_name_info[0] == '\0') {
