From eae89e8432d993417e6f0e17ae519b4e02f72c9b Mon Sep 17 00:00:00 2001
From: Krzesimir Nowak <knowak@microsoft.com>
Date: Fri, 7 Nov 2025 12:06:05 +0100
Subject: [PATCH 2/2] Patch yaml-cpp after fetching to fix cmake issues

Signed-off-by: Krzesimir Nowak <knowak@microsoft.com>
---
 CMakeLists.txt | 1 +
 1 file changed, 1 insertion(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 753b081..03b2ab9 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -53,6 +53,7 @@ if (NOT yaml-cpp_FOUND)
     yaml-cpp
     GIT_REPOSITORY https://github.com/jbeder/yaml-cpp.git
     GIT_TAG 0.8.0
+    PATCH_COMMAND patch -p1 -i @FILESDIR@/yaml-cpp-cmake-4.0.patch
   )
   FetchContent_MakeAvailable(yaml-cpp)
 endif()
-- 
2.51.0

