From 0ee2c928de51ef56376c4ab6528ca1e23c13c4a7 Mon Sep 17 00:00:00 2001
From: Krzesimir Nowak <knowak@microsoft.com>
Date: Fri, 8 Aug 2025 16:29:52 +0200
Subject: [PATCH] fix

---
 scsi.cpp | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/scsi.cpp b/scsi.cpp
index d8c27a9..4c7f480 100644
--- a/scsi.cpp
+++ b/scsi.cpp
@@ -183,7 +183,7 @@ int tcmu_emulate_evpd_inquiry(
 	{
 		char data[512];
 		char *ptr, *p, *wwn;
-		size_t len, used = 0;
+		size_t len, used = 4;
 		uint16_t *tot_len = (uint16_t*) &data[2];
 		uint32_t padding;
 		bool next;
@@ -207,7 +207,7 @@ int tcmu_emulate_evpd_inquiry(
 
 		ptr[3] = 8 + len + 1;
 		used += (uint8_t)ptr[3] + 4;
-		ptr += used;
+		ptr += (uint8_t)ptr[3] + 4;
 
 		/* 2/5: NAA binary */
 		ptr[0] = 1; /* code set: binary */
@@ -340,9 +340,9 @@ int tcmu_emulate_evpd_inquiry(
 finish_page83:
 		/* Done with descriptor list */
 
-		*tot_len = htobe16(used);
+		*tot_len = htobe16(used - 4);
 
-		tcmu_memcpy_into_iovec(iovec, iov_cnt, data, used + 4);
+		tcmu_memcpy_into_iovec(iovec, iov_cnt, data, used);
 
 		free(wwn);
 		wwn = NULL;
-- 
2.49.1

