--- esync.py	2006-04-23 19:52:23.000000000 -0400
+++ esync.py.new	2006-04-24 00:33:55.217967750 -0400
@@ -22,7 +22,7 @@
 
 from common import needdbversion
 
-syncprogram =   "/usr/bin/emerge sync"
+syncprogram =   "EMERGE_DEFAULT_OPTS=\"\" /usr/bin/emerge sync"
 logfile_sync =  "/var/log/emerge-sync.log"
 tmp_prefix =    "/tmp/esync"
 
@@ -72,9 +72,9 @@
     if arg in ("-h", "--help"):
         usage()
     elif arg in ("-w", "--webrsync"):
-        syncprogram = "/usr/sbin/emerge-webrsync"
+        syncprogram = "EMERGE_DEFAULT_OPTS=\"\" /usr/sbin/emerge-webrsync"
     elif arg in ("-m", "--metadata"):
-        syncprogram = "/usr/bin/emerge --metadata"
+        syncprogram = "EMERGE_DEFAULT_OPTS=\"\" /usr/bin/emerge --metadata"
     elif arg in ("-n", "--nocolor"):
         eoptions = "-n"
         nocolor()
