diff -ru esearch.orig/esearch.py esearch/esearch.py
--- esearch.orig/esearch.py	2008-12-30 21:14:03.000000000 -0600
+++ esearch/esearch.py	2008-12-30 21:10:22.000000000 -0600
@@ -13,7 +13,10 @@
 sys.path.insert(0, "/usr/lib/portage/pym")
 sys.path.insert(0, "/usr/lib/esearch")
 
-from output import bold, red, green, darkgreen, turquoise, nocolor
+try:
+    from portage.output import bold, red, green, darkgreen, turquoise, nocolor
+except ImportError:
+    from output import bold, red, green, darkgreen, turquoise, nocolor, blue
 from os.path import exists
 import re
 
@@ -128,7 +131,10 @@
     elif arg in ("-v", "--verbose"):
         import string
         from portage import portdb, best, settings
-        from output import blue
+        try:
+            from portage.output import blue
+        except ImportError:
+            from output import blue
         from common import version
         outputm = VERBOSE
     elif arg in ("-e", "--ebuild"):
diff -ru esearch.orig/esync.py esearch/esync.py
--- esearch.orig/esync.py	2008-12-30 21:14:05.000000000 -0600
+++ esearch/esync.py	2008-12-30 20:58:16.000000000 -0600
@@ -18,7 +18,10 @@
 sys.path.insert(0, "/usr/lib/portage/pym")
 
 import portage
-from output import red, green, bold, darkgreen, nocolor, xtermTitle
+try:
+    from portage.output import red, green, bold, darkgreen, nocolor, xtermTitle
+except ImportError:
+    from output import red, green, bold, darkgreen, nocolor, xtermTitle
 
 from common import needdbversion
 
diff -ru esearch.orig/eupdatedb.py esearch/eupdatedb.py
--- esearch.orig/eupdatedb.py	2008-12-30 21:14:08.000000000 -0600
+++ esearch/eupdatedb.py	2008-12-30 20:58:16.000000000 -0600
@@ -20,7 +20,10 @@
 sys.path.insert(0, "/usr/lib/esearch")
 
 import portage
-from output import red, darkgreen, green, bold, nocolor
+try:
+    from portage.output import red, darkgreen, green, bold, nocolor
+except ImportError:
+    from output import red, darkgreen, green, bold, nocolor
 
 from common import needdbversion, version
 
