diff -ur esearch-0.7.1/en/esync.1 esearch-0.7.1-patched/en/esync.1
--- en/esync.1	2005-02-13 18:12:03.000000000 +0100
+++ en/esync.1	2006-06-05 13:37:40.000000000 +0200
@@ -20,6 +20,9 @@
 .B \-\-webrsync, \-w
 Use 'emerge-webrsync' instead of 'emerge sync'
 .TP
+.B \-\-metadata, \-m
+Use 'emerge --metadata' instead of 'emerge sync'
+.TP
 .B \-\-nocolor, \-n
 Don't use ANSI codes for colored output
 .TP
diff -ur esearch-0.7.1/it/esync.1 esearch-0.7.1-patched/it/esync.1
--- it/esync.1	2005-02-13 17:23:38.000000000 +0100
+++ it/esync.1	2006-06-05 13:40:13.000000000 +0200
@@ -20,6 +20,9 @@
 .B \-\-webrsync, \-w
 Usa 'emerge-webrsync' invece di 'emerge sync'.
 .TP
+.B \-\-metadata, \-m
+Usa 'emerge --metadata' invece di 'emerge sync'.
+.TP
 .B \-\-nocolor, \-n
 Non usa i codici colore ANSI per la stampa.
 .TP
diff -ur esearch-0.7.1/esync.py esearch-0.7.1-patched/esync.py
--- esync.py	2005-02-13 17:53:38.000000000 +0100
+++ esync.py	2006-06-05 13:37:40.000000000 +0200
@@ -44,6 +44,9 @@
     print darkgreen("  --webrsync") + ", " + darkgreen("-w")
     print "    Use 'emerge-webrsync' instead of 'emerge sync'"
     print ""
+    print darkgreen("  --metadata") + ", " + darkgreen("-m")
+    print "    Use 'emerge --metadata' instead of 'emerge sync'"
+    print ""
     print darkgreen("  --nocolor") + ", " + darkgreen("-n")
     print "    Don't use ANSI codes for colored output"
     print ""
@@ -57,7 +60,7 @@
     sys.exit(0)
 
 try:
-    opts = getopt(sys.argv[1:], "hwnvs", ["help", "webrsync", "nocolor", "verbose", "nospinner"])
+    opts = getopt(sys.argv[1:], "hwmnvs", ["help", "webrsync", "nocolor", "verbose", "metadata", "nospinner"])
 except GetoptError, error:
     print red(" * Error:"), error, "(see", darkgreen("--help"), "for all options)"
     print
@@ -70,6 +73,8 @@
         usage()
     elif arg in ("-w", "--webrsync"):
         syncprogram = "/usr/sbin/emerge-webrsync"
+    elif arg in ("-m", "--metadata"):
+        syncprogram = "/usr/bin/emerge --metadata"
     elif arg in ("-n", "--nocolor"):
         eoptions = "-n"
         nocolor()
