--- a/esearch.py	2006-04-23 19:48:38.088214000 -0400
+++ b/esearch.py	2006-04-23 19:33:21.482929750 -0400
@@ -222,7 +222,11 @@
 
                 if outputm == VERBOSE:
                     mpv = best(portdb.xmatch("match-all", pkg[1]))
-                    iuse_split = string.split(portdb.aux_get(pkg[1] + "-" +  pkg[3], ["IUSE"])[0], " ")
+                    try:
+                        iuse_split = string.split(portdb.aux_get(pkg[1] + "-" +  pkg[3], ["IUSE"])[0], " ")
+                    except KeyError, e:
+                        print "Package %s is no longer in the portage tree." % pkg[1] + "-" + pkg[3]
+                        continue
                     iuse_split.sort()
                     iuse = ""
 
