diff --git a/src/luks/dracut/clevis-pin-tang/module-setup.sh.in b/src/luks/dracut/clevis-pin-tang/module-setup.sh.in
index 929b878..c48e282 100755
--- a/src/luks/dracut/clevis-pin-tang/module-setup.sh.in
+++ b/src/luks/dracut/clevis-pin-tang/module-setup.sh.in
@@ -19,7 +19,7 @@
 #
 
 depends() {
-    echo clevis network
+    echo clevis
     return 0
 }
 
diff --git a/src/luks/dracut/clevis/module-setup.sh.in b/src/luks/dracut/clevis/module-setup.sh.in
index dbce790..c9581db 100755
--- a/src/luks/dracut/clevis/module-setup.sh.in
+++ b/src/luks/dracut/clevis/module-setup.sh.in
@@ -48,7 +48,6 @@ install() {
     fi
 
     inst_multiple \
-        /etc/services \
         clevis-luks-common-functions \
         grep sed cut \
         clevis-decrypt \
