--- pciutils-2.2.7/update-pciids.sh
+++ pciutils-2.2.7/update-pciids.sh
@@ -72,4 +72,19 @@
 	rm -f ${DEST%.gz} ${DEST%.gz}.old
 fi
 
+# if the gzip version exists, but we are not using compressed mode, 
+# erase the gzip version, as it is old and crufty
+if [ -z "$PCI_COMPRESSED_IDS" ] && [ -f $DEST.gz ]; then
+	rm -f ${DEST}.gz
+fi
+
+# Lastly, because HAL still reads the plaintext version only and 
+# has not yet migrated to using a library we leave the plaintext 
+# version in addition to the gzip version.
+# They need to migrate because the pci.ids format will be changing.
+if [ ${DEST%.gz} != ${DEST} ] ; then
+	gzip -d <${DEST} >${DEST%.gz}
+fi
+chmod a+r ${DEST} ${DEST%.gz}
+
 ${quiet} || echo "Done."
