http://bugs.gentoo.org/212784

--- libxslt-1.1.22/work/libxslt-1.1.22/python/Makefile.am 
+++ libxslt-1.1.23/work/libxslt-1.1.23/python/Makefile.am 
@@ -42,9 +42,9 @@
 	cat $(srcdir)/libxsl.py $(srcdir)/libxsltclass.py > libxslt.py
 
 install-data-local:
-	$(mkinstalldirs) $(DESTDIR)$(pythondir)
+	-$(mkinstalldirs) $(DESTDIR)$(pythondir)
 	@INSTALL@ -m 0644 libxslt.py $(DESTDIR)$(pythondir)
-	$(mkinstalldirs) $(DESTDIR)$(DOCS_DIR)
+	-$(mkinstalldirs) $(DESTDIR)$(DOCS_DIR)
 	@(for doc in $(DOCS) ; \
 	   do @INSTALL@ -m 0644 $(srcdir)/$$doc $(DESTDIR)$(DOCS_DIR) ; done)
 
