diff -Nru lrzsz-0.12.20.orig/src/Makefile.am lrzsz-0.12.20/src/Makefile.am
--- lrzsz-0.12.20.orig/src/Makefile.am	1998-12-28 10:38:47.000000000 +0200
+++ lrzsz-0.12.20/src/Makefile.am	2004-11-19 12:34:16.766260535 +0200
@@ -15,21 +15,21 @@
 DEFS = -DNFGVMIN @DEFS@ -DLOCALEDIR=\"$(localedir)\"
 
 install-exec-local:
-	rm -f $(DESTDIR)/$(bindir)/`echo lsb | sed -e '$(transform)'`
-	ln $(DESTDIR)/$(bindir)/`echo lsz |sed -e '$(transform)'` \
-		$(DESTDIR)/$(bindir)/`echo lsb |sed -e '$(transform)'` 
-	rm -f $(bindir)/`echo lsx | sed -e '$(transform)'`
-	ln $(DESTDIR)/$(bindir)/`echo lsz |sed -e '$(transform)'` \
-		$(DESTDIR)/$(bindir)/`echo lsx |sed -e '$(transform)'` 
-	rm -f $(bindir)/`echo lrb | sed -e '$(transform)'`
-	ln $(DESTDIR)/$(bindir)/`echo lrz |sed -e '$(transform)'` \
-		$(DESTDIR)/$(bindir)/`echo lrb |sed -e '$(transform)'` 
-	rm -f $(bindir)/`echo lrx | sed -e '$(transform)'`
-	ln $(DESTDIR)/$(bindir)/`echo lrz |sed -e '$(transform)'` \
-		$(DESTDIR)/$(bindir)/`echo lrx |sed -e '$(transform)'` 
-#	rm -f $(bindir)/`echo lrzszbug | sed -e '$(transform)'`
-#	ln $(bindir)/`echo lrzszbug |sed -e '$(transform)'` \
-#		$(bindir)/`echo lrzszbug |sed -e '$(transform)'` 
+	rm -f $(DESTDIR)/$(bindir)/`echo lsb | sed -e '$(transform)'`; \
+		ln $(DESTDIR)/$(bindir)/`echo lsz |sed -e '$(transform)'` \
+			$(DESTDIR)/$(bindir)/`echo lsb |sed -e '$(transform)'` 
+	rm -f $(bindir)/`echo lsx | sed -e '$(transform)'`; \
+		ln $(DESTDIR)/$(bindir)/`echo lsz |sed -e '$(transform)'` \
+			$(DESTDIR)/$(bindir)/`echo lsx |sed -e '$(transform)'` 
+	rm -f $(bindir)/`echo lrb | sed -e '$(transform)'`; \
+		ln $(DESTDIR)/$(bindir)/`echo lrz |sed -e '$(transform)'` \
+			$(DESTDIR)/$(bindir)/`echo lrb |sed -e '$(transform)'` 
+	rm -f $(bindir)/`echo lrx | sed -e '$(transform)'`; \
+		ln $(DESTDIR)/$(bindir)/`echo lrz |sed -e '$(transform)'` \
+			$(DESTDIR)/$(bindir)/`echo lrx |sed -e '$(transform)'` 
+#	rm -f $(bindir)/`echo lrzszbug | sed -e '$(transform)'`; \
+#		ln $(bindir)/`echo lrzszbug |sed -e '$(transform)'` \
+#			$(bindir)/`echo lrzszbug |sed -e '$(transform)'` 
 
 uninstall-local:
 	rm -f $(DESTDIR)/$(bindir)/`echo lsb |sed -e '$(transform)'` 
diff -Nru lrzsz-0.12.20.orig/src/Makefile.in lrzsz-0.12.20/src/Makefile.in
--- lrzsz-0.12.20.orig/src/Makefile.in	1998-12-30 18:31:46.000000000 +0200
+++ lrzsz-0.12.20/src/Makefile.in	2004-11-19 12:33:45.294953025 +0200
@@ -369,21 +369,21 @@
 
 
 install-exec-local:
-	rm -f $(DESTDIR)/$(bindir)/`echo lsb | sed -e '$(transform)'`
-	ln $(DESTDIR)/$(bindir)/`echo lsz |sed -e '$(transform)'` \
-		$(DESTDIR)/$(bindir)/`echo lsb |sed -e '$(transform)'` 
-	rm -f $(bindir)/`echo lsx | sed -e '$(transform)'`
-	ln $(DESTDIR)/$(bindir)/`echo lsz |sed -e '$(transform)'` \
-		$(DESTDIR)/$(bindir)/`echo lsx |sed -e '$(transform)'` 
-	rm -f $(bindir)/`echo lrb | sed -e '$(transform)'`
-	ln $(DESTDIR)/$(bindir)/`echo lrz |sed -e '$(transform)'` \
-		$(DESTDIR)/$(bindir)/`echo lrb |sed -e '$(transform)'` 
-	rm -f $(bindir)/`echo lrx | sed -e '$(transform)'`
-	ln $(DESTDIR)/$(bindir)/`echo lrz |sed -e '$(transform)'` \
-		$(DESTDIR)/$(bindir)/`echo lrx |sed -e '$(transform)'` 
-#	rm -f $(bindir)/`echo lrzszbug | sed -e '$(transform)'`
-#	ln $(bindir)/`echo lrzszbug |sed -e '$(transform)'` \
-#		$(bindir)/`echo lrzszbug |sed -e '$(transform)'` 
+	rm -f $(DESTDIR)/$(bindir)/`echo lsb | sed -e '$(transform)'`; \
+		ln $(DESTDIR)/$(bindir)/`echo lsz |sed -e '$(transform)'` \
+			$(DESTDIR)/$(bindir)/`echo lsb |sed -e '$(transform)'` 
+	rm -f $(bindir)/`echo lsx | sed -e '$(transform)'`; \
+		ln $(DESTDIR)/$(bindir)/`echo lsz |sed -e '$(transform)'` \
+			$(DESTDIR)/$(bindir)/`echo lsx |sed -e '$(transform)'` 
+	rm -f $(bindir)/`echo lrb | sed -e '$(transform)'`; \
+		ln $(DESTDIR)/$(bindir)/`echo lrz |sed -e '$(transform)'` \
+			$(DESTDIR)/$(bindir)/`echo lrb |sed -e '$(transform)'` 
+	rm -f $(bindir)/`echo lrx | sed -e '$(transform)'`; \
+		ln $(DESTDIR)/$(bindir)/`echo lrz |sed -e '$(transform)'` \
+			$(DESTDIR)/$(bindir)/`echo lrx |sed -e '$(transform)'` 
+#	rm -f $(bindir)/`echo lrzszbug | sed -e '$(transform)'`; \
+#		ln $(bindir)/`echo lrzszbug |sed -e '$(transform)'` \
+#			$(bindir)/`echo lrzszbug |sed -e '$(transform)'` 
 
 uninstall-local:
 	rm -f $(DESTDIR)/$(bindir)/`echo lsb |sed -e '$(transform)'` 
