diff -r -u libsemanage-3.6-patched/src/semanage.conf libsemanage-3.6-flatcar/src/semanage.conf
--- libsemanage-3.6-patched/src/semanage.conf	2024-02-21 14:33:06.055611733 +0100
+++ libsemanage-3.6-flatcar/src/semanage.conf	2024-02-21 14:42:13.566991009 +0100
@@ -58,7 +58,7 @@
 # 1-9 when compressing.  The higher the number,
 # the more memory is traded off for disk space.
 # Set to 0 to disable bzip2 compression.
-bzip-blocksize=0
+bzip-blocksize=1
 
 # Reduce memory usage for bzip2 compression and
 # decompression of modules in the module store.
