diff --git a/configure.ac b/configure.ac
index 4000a38aa8..b5917dd618 100644
--- a/configure.ac
+++ b/configure.ac
@@ -14,6 +14,13 @@ AC_INIT([Lustre], m4_esyscmd_s([./LUSTRE-VERSION-GEN]),
 
 AC_CONFIG_AUX_DIR([config])
 AC_CONFIG_MACRO_DIR([config])
+m4_ifdef([AC_CONFIG_MACRO_DIRS], [
+AC_CONFIG_MACRO_DIRS([libcfs/autoconf lnet/autoconf lustre/autoconf])
+], [
+AC_CONFIG_MACRO_DIR([libcfs/autoconf])
+AC_CONFIG_MACRO_DIR([lnet/autoconf])
+AC_CONFIG_MACRO_DIR([lustre/autoconf])
+])
 
 LC_CONFIG_SRCDIR
 
