From c3e6b575bdb267c3500e95934f83247ccc7792dc Mon Sep 17 00:00:00 2001
From: Mathieu Tortuyaux <mtortuyaux@microsoft.com>
Date: Wed, 25 May 2022 10:10:45 +0200
Subject: [PATCH 02/15] mod: add ign-converter dependency

Signed-off-by: Mathieu Tortuyaux <mtortuyaux@microsoft.com>
---
 go.mod | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/go.mod b/go.mod
index e5a47737..9b5da9f3 100644
--- a/go.mod
+++ b/go.mod
@@ -10,6 +10,8 @@ require (
 	github.com/coreos/go-semver v0.3.0
 	github.com/coreos/go-systemd/v22 v22.0.0
 	github.com/coreos/vcontext v0.0.0-20211021162308-f1dbbca7bef4
+	github.com/flatcar/ign-converter v0.2.0
+	github.com/flatcar/ignition v0.36.2-0.20220221101037-de4e6cc9bbba
 	github.com/google/renameio v0.1.0
 	github.com/google/uuid v1.1.1
 	github.com/pin/tftp v2.1.0+incompatible
@@ -21,7 +23,9 @@ require (
 	golang.org/x/net v0.0.0-20200602114024-627f9648deb9
 	golang.org/x/oauth2 v0.0.0-20200107190931-bf48bf16ab8d
 	golang.org/x/sys v0.0.0-20200610111108-226ff32320da
+	golang.org/x/text v0.3.7 // indirect
 	golang.org/x/tools v0.0.0-20200610160956-3e83d1e96d0e // indirect
 	google.golang.org/api v0.26.0
 	google.golang.org/genproto v0.0.0-20200610104632-a5b850bcf112 // indirect
+	gopkg.in/yaml.v2 v2.4.0 // indirect
 )
-- 
2.35.1

