--- ssmtp.orig.conf	2004-07-23 00:58:48.000000000 -0500
+++ ssmtp.conf	2008-07-14 14:11:34.000000000 -0500
@@ -2,10 +2,13 @@
 # /etc/ssmtp.conf -- a config file for sSMTP sendmail.
 #
 
-# The person who gets all mail for userids < 1000
+# The person who gets all mail for userids < MinUserId
 # Make this empty to disable rewriting.
 root=postmaster
 
+# All mail delivered to userid >= MinUserId goes to user, not root.
+#MinUserId=1000
+
 # The place where the mail goes. The actual machine name is required
 # no MX records are consulted. Commonly mailhosts are named mail.domain.com
 # The example will fit if you are in domain.com and your mailhub is so named.
