From 35618961c9047ddc088bc43274be40e77858dfe7 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Kai=20L=C3=BCke?= <kai@kinvolk.io>
Date: Tue, 24 Sep 2019 18:22:06 +0200
Subject: [PATCH] Use rpcgen from host to allow cross-compilation

---
 rpcsvc/Makefile.am | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/rpcsvc/Makefile.am b/rpcsvc/Makefile.am
index 83867c0..c1610a7 100644
--- a/rpcsvc/Makefile.am
+++ b/rpcsvc/Makefile.am
@@ -12,5 +12,5 @@ nodist_rpcsvc_HEADERS = klm_prot.h nlm_prot.h rstat.h spray.h \
 	nfs_prot.h rquota.h sm_inter.h
 
 %.h: %.x
-	$(top_builddir)/rpcgen/rpcgen -h -o $@ $<
+	rpcgen -h -o $@ $<
 
-- 
2.21.0

