From b4ce1af6005f6137774ba69fb1db5b320a853513 Mon Sep 17 00:00:00 2001
From: Nick Owens <nick.owens@coreos.com>
Date: Tue, 2 Jun 2015 18:22:32 -0700
Subject: [PATCH 2/6] networkd: default to "kernel" IPForwarding setting

---
 src/network/networkd-network.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/network/networkd-network.c b/src/network/networkd-network.c
index 39ea4eddd0..9780f920f1 100644
--- a/src/network/networkd-network.c
+++ b/src/network/networkd-network.c
@@ -464,6 +464,7 @@ int network_load_one(Manager *manager, OrderedHashmap **networks, const char *fi
                 .link_local = _ADDRESS_FAMILY_INVALID,
                 .ipv6ll_address_gen_mode = _IPV6_LINK_LOCAL_ADDRESS_GEN_MODE_INVALID,
 
+                .ip_forward = _ADDRESS_FAMILY_INVALID,
                 .ipv4_accept_local = -1,
                 .ipv4_route_localnet = -1,
                 .ipv6_privacy_extensions = IPV6_PRIVACY_EXTENSIONS_NO,
-- 
2.25.1

