From bbc8ff28a7f0208b6b1020d0d2ca900cdfb5a42f Mon Sep 17 00:00:00 2001
From: Krzesimir Nowak <knowak@microsoft.com>
Date: Fri, 24 Oct 2025 11:06:57 +0200
Subject: [PATCH 08/13] units: Make multi-user.target the default target

Signed-off-by: Krzesimir Nowak <knowak@microsoft.com>
---
 units/meson.build | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/units/meson.build b/units/meson.build
index 12cf47d797..58de76dca9 100644
--- a/units/meson.build
+++ b/units/meson.build
@@ -47,10 +47,7 @@ units = [
           'file' : 'getty@.service.in',
           'symlinks' : ['autovt@.service'],
         },
-        {
-          'file' : 'graphical.target',
-          'symlinks' : ['default.target'],
-        },
+        { 'file' : 'graphical.target' },
         { 'file' : 'halt.target' },
         {
           'file' : 'hibernate.target',
@@ -142,7 +139,10 @@ units = [
           'conditions' : ['ENABLE_MACHINED'],
         },
         { 'file' : 'modprobe@.service' },
-        { 'file' : 'multi-user.target' },
+        {
+          'file' : 'multi-user.target',
+          'symlinks' : ['default.target'],
+        },
         {
           'file' : 'systemd-mute-console.socket',
           'symlinks' : ['sockets.target.wants/']
-- 
2.52.0

