From 3c13363e4b3f2e5bcc762a71460d84b93452f53f Mon Sep 17 00:00:00 2001
From: Krzesimir Nowak <knowak@microsoft.com>
Date: Fri, 24 Oct 2025 11:06:57 +0200
Subject: [PATCH] units: Make multi-user.target the default target

Signed-off-by: Krzesimir Nowak <knowak@microsoft.com>
Signed-off-by: Kai Lueke <kailuke@microsoft.com>
---
 units/meson.build | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/units/meson.build b/units/meson.build
index 4f47a3b2bd..9663e21e0c 100644
--- a/units/meson.build
+++ b/units/meson.build
@@ -48,8 +48,7 @@ units = [
           'symlinks' : ['autovt@.service'],
         },
         {
-          'file' : 'graphical.target',
-          'symlinks' : ['default.target'],
+          'file' : 'graphical.target'
         },
         { 'file' : 'halt.target' },
         {
@@ -142,7 +141,9 @@ units = [
           'conditions' : ['ENABLE_MACHINED'],
         },
         { 'file' : 'modprobe@.service' },
-        { 'file' : 'multi-user.target' },
+        { 'file' : 'multi-user.target' ,
+          'symlinks' : ['default.target']
+        },
         { 'file' : 'network-online.target' },
         { 'file' : 'network-pre.target' },
         { 'file' : 'network.target' },
-- 
2.51.0
