From 3f059a6a123dd62257f224b9af7701078783060e Mon Sep 17 00:00:00 2001
From: Matthias Clasen <mclasen@redhat.com>
Date: Sat, 22 Jan 2011 04:41:12 +0000
Subject: Remove a test that seems to fail depending on time of day

---
diff --git a/glib/tests/gdatetime.c b/glib/tests/gdatetime.c
index 01f08de..619bb02 100644
--- a/glib/tests/gdatetime.c
+++ b/glib/tests/gdatetime.c
@@ -804,7 +804,7 @@ GDateTime *__dt = g_date_time_new_local (2009, 10, 24, 0, 0, 0);\
   TEST_PRINTF_TIME (13, 13, 13, "%r", "01:13:13 PM");
   TEST_PRINTF ("%R", "00:00");
   TEST_PRINTF_TIME (13, 13, 31, "%R", "13:13");
-  TEST_PRINTF ("%s", t_str);
+  //TEST_PRINTF ("%s", t_str);
   TEST_PRINTF ("%S", "00");
   TEST_PRINTF ("%t", "	");
   TEST_PRINTF ("%W", "42");
--
cgit v0.8.3.1
